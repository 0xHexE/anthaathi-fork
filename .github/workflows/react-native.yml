name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
