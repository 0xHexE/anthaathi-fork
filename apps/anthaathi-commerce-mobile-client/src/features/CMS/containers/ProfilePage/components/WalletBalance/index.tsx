import {View, Pressable} from 'react-native';
import React from 'react';
import {Text} from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {useIntl} from 'react-intl';
import {ProfilePageComponentType} from '../../../../types/common';

export interface WalletBalanceProps {
  title: string;
  balance: number;
  currency: string;
  handlePress?: () => void;
  buttonTitle: string;
}

const WalletBalance = (props: WalletBalanceProps) => {
  const intl = useIntl();
  return (
    <View testID="walletBalance">
      <Pressable
        testID="handlePress"
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          marginHorizontal: 10,
          height: 60,
        }}
        onPress={props.handlePress}>
        <View
          style={{
            width: '80%',
            justifyContent: 'center',
            borderWidth: 1,
            borderColor: '#008D3E',
            borderTopLeftRadius: 2,
            borderBottomLeftRadius: 2,
          }}>
          <Text
            style={{
              fontSize: 12,
              color: '#808080',
              fontWeight: '400',
              marginLeft: 10,
            }}>
            {props.title}
          </Text>
          <Text
            style={{
              fontSize: 14,
              color: '#008D3E',
              fontWeight: '700',
              marginLeft: 10,
            }}>
            {intl.formatNumber(props.balance, {
              style: 'currency',
              currency: props.currency,
            })}
          </Text>
        </View>
        <View
          style={{
            width: '20%',
            backgroundColor: '#008D3E',
            alignItems: 'center',
            justifyContent: 'center',
            borderWidth: 1,
            borderColor: '#008D3E',
            borderTopRightRadius: 2,
            borderBottomRightRadius: 2,
          }}>
          <Icon name="clipboard-outline" color="#ffffff" size={18} />
          <Text style={{fontSize: 10, fontWeight: '400', color: '#ffffff'}}>
            {props.buttonTitle}
          </Text>
        </View>
      </Pressable>
    </View>
  );
};

export default WalletBalance;

export const WalletBalanceCMSInput = {
  _component: ProfilePageComponentType.WalletBalance,
  component: WalletBalance,
};
