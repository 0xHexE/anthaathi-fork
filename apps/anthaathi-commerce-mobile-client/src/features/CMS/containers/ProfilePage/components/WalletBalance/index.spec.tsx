import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import WalletBalance from './index';

describe('WalletBalance', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <WalletBalance
            title="Wallet Balance"
            balance={0}
            currency="USD"
            buttonTitle="Top-up"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('walletBalance')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <WalletBalance
            title="Wallet Balance"
            balance={0}
            currency="USD"
            buttonTitle="Top-up"
            handlePress={onpress}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('handlePress')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have WalletBalance title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <WalletBalance
            title="Wallet Balance"
            balance={0}
            currency="USD"
            buttonTitle="Top-up"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Wallet Balance')).toBeTruthy();
  });
});
