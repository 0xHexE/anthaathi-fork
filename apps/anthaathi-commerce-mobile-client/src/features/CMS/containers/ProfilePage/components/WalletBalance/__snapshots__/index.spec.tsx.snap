// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletBalance should render the item 1`] = `
<View
  testID="walletBalance"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "marginHorizontal": 10,
      }
    }
    testID="handlePress"
  >
    <View
      style={
        Object {
          "borderBottomLeftRadius": 2,
          "borderColor": "#008D3E",
          "borderTopLeftRadius": 2,
          "borderWidth": 1,
          "justifyContent": "center",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#808080",
              "fontSize": 12,
              "fontWeight": "400",
              "marginLeft": 10,
            },
          ]
        }
      >
        Wallet Balance
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#008D3E",
              "fontSize": 14,
              "fontWeight": "700",
              "marginLeft": 10,
            },
          ]
        }
      >
        $0.00
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#008D3E",
          "borderBottomRightRadius": 2,
          "borderColor": "#008D3E",
          "borderTopRightRadius": 2,
          "borderWidth": 1,
          "justifyContent": "center",
          "width": "20%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°…Œ
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#ffffff",
              "fontSize": 10,
              "fontWeight": "400",
            },
          ]
        }
      >
        Top-up
      </Text>
    </View>
  </View>
</View>
`;
